src/platform/windows.rs

```diff
@@ -56,6 +56,38 @@ pub trait EventLoopBuilderExtWindows {
     /// # }
     /// ```
     fn with_dpi_aware(&mut self, dpi_aware: bool) -> &mut Self;
+
+    /// A callback to be executed before dispatching a win32 message to the window procedure.
+    /// Return true to disable winit's internal message dispatching.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use  windows_sys::Win32::UI::WindowsAndMessaging::{ACCEL, CreateAcceleratorTableW, TranslateAcceleratorW, DispatchMessageW, TranslateMessage, MSG};
+    /// use winit::event_loop::EventLoopBuilder;
+    /// #[cfg(target_os = "windows")]
+    /// use winit::platform::windows::EventLoopBuilderExtWindows;
+    ///
+    /// let mut builder = EventLoopBuilder::new();
+    /// #[cfg(target_os = "windows")]
+    /// builder.with_msg_hook(Box::new(|mut msg|{
+    ///     let msg = msg as *mut MSG;
+    /// #   let accels: Vec<ACCEL> = Vec::new();
+    ///     unsafe {
+    ///         let translated = TranslateAcceleratorW(
+    ///             (*msg).hwnd,
+    ///             CreateAcceleratorTableW(accels.as_ptr() as _, 1),
+    ///             msg,
+    ///         ) == 1;
+    ///         if !translated {
+    ///             TranslateMessage(msg);
+    ///             DispatchMessageW(msg);
+    ///         }
+    ///     }
+    ///     true
+    /// }));
+    /// ```
+    fn with_msg_hook(&mut self, callback: Box<dyn FnMut(*mut c_void) -> bool>) -> &mut Self;
```

Comment:
I think we can also hide the `Box` in the interface by making the funktion generic and box it internally.
```rust
fn with_msg_hook<F>(&mut self, callback: F) where F: FnMut(*const c_void) -> bool + 'static
```